orm/idgen_bucket.go

```diff
@@ -0,0 +1,51 @@
+package orm
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+// IDGenerator defines an interface for custom id generators
+type IDGenerator interface {
+	// NextVal returns a new unique ID key
+	NextVal(db weave.KVStore, obj CloneableData) ([]byte, error)
```

Comment:
Why do you take obj here? If the nextval is just based on the bucket?

Reply:
I had the idea of a generic IDGenerator interface that supports natural keys as well.